Analysis/Tasks/PWGHF/HFCandidateCreatorXicc.cxx

```diff
@@ -0,0 +1,280 @@
+                         pvecxic[0], pvecxic[1], pvecxic[2],
+                         pvecpion[0], pvecpion[1], pvecpion[2],
+                         impactParameter0.getY(), impactParameter1.getY(),
+                         std::sqrt(impactParameter0.getSigmaY2()), std::sqrt(impactParameter1.getSigmaY2()),
+                         xicCand.globalIndex(), trackpion.globalIndex(),
+                         hfFlag);
+        // calculate invariant mass
+        auto arrayMomenta = array{pvecxic, pvecpion};
+        massXicc = RecoDecay::M(std::move(arrayMomenta), array{massXic, massPi});
+        hmassXicc->Fill(massXicc);
+      } // if on selected Xicc
+    }   // loop over candidates
+  }     // end of process
+};      //end of struct
+
+/// Extends the base table with expression columns.
+struct HFCandidateCreatorXiccExpressions {
+  Spawns<aod::HfCandXiccExt> rowCandidateXicc;
+  void init(InitContext const&) {}
+};
+
+/// Performs MC matching.
+struct HFCandidateCreatorXiccMC {
+  Produces<aod::HfCandXiccMCRec> rowMCMatchRec;
+  Produces<aod::HfCandXiccMCGen> rowMCMatchGen;
+
+  void process(aod::HfCandXicc const& candidates,
+               aod::HfCandProng3,
+               aod::BigTracksMC const& tracks,
+               aod::McParticles const& particlesMC)
+  {
+    int indexRec = -1;
+    int8_t sign = 0;
+    int8_t flag = 0;
+    int8_t origin = 0;
+    int8_t channel = 0;
+
+    // Match reconstructed candidates.
+    for (auto& candidate : candidates) {
+      //Printf("New rec. candidate");
+      flag = 0;
+      origin = 0;
+      channel = 0;
+
+      auto xicCand = candidate.index0();
+      auto arrayDaughters = array{xicCand.index0_as<aod::BigTracksMC>(),
+                                  xicCand.index1_as<aod::BigTracksMC>(),
+                                  xicCand.index2_as<aod::BigTracksMC>(),
+                                  candidate.index1_as<aod::BigTracksMC>()};
+
+      // Ξcc±± → p± K∓ π± π±
+      //Printf("Checking Ξcc±± → p± K∓ π± π±");
+      //FIXME Be aware that at the moment, you are not considering any resonant channel
+      indexRec = RecoDecay::getMatchedMCRec(particlesMC, arrayDaughters, pdg::Code::kXiCCPlusPlus, array{+kProton, -kKPlus, +kPiPlus, +kPiPlus}, true, &sign, 3);
+      if (indexRec > -1) {
+        flag = 1 << XiccToXicPi;
+      }
+
+      // Check whether the particle is non-prompt (from a b quark).
+      if (flag != 0) {
+        auto particle = particlesMC.iteratorAt(indexRec);
+        origin = (RecoDecay::getMother(particlesMC, particle, 5, true) > -1 ? NonPrompt : Prompt);
+      }
+
+      rowMCMatchRec(flag, origin, channel);
+    }
+
+    // Match generated particles.
+    for (auto& particle : particlesMC) {
+      //Printf("New gen. candidate");
+      flag = 0;
+      origin = 0;
+      channel = 0;
+      // Xicc → Xic + π+
+      if (RecoDecay::isMatchedMCGen(particlesMC, particle, pdg::Code::kXiCCPlusPlus, array{4232, +kPiPlus}, true)) {
```

Comment:
`4232` -> `pdg::Code::kXiCPlus`

Reply:
It doesnt like this. See below:

 24   251 |       if (RecoDecay::isMatchedMCGen(particlesMC, particle, pdg::Code::kXiCCPlusPlus, array{pdg::Code::kXiCPlus, +kPiPlus}, true)) {
 25       |                                                                                                                         ^
 26 /home/ginnocen/alice/sw/SOURCES/O2/xicc/0/Analysis/Tasks/PWGHF/HFCandidateCreatorXicc.cxx:251:121: error: no matching function for call to 'array(o2::analysis::pdg::Code, int)'

Reply:
thanks, done.